### PR TITLE
feat(wasm): BitmapImage now supports scale qualifiers

### PR DESCRIPTION
**Pull request by ebariche**
GitHub Issue (If applicable): Fixes unoplatform/private#153

## PR Type

What kind of change does this PR introduce?
- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+						return new ImageData() { Kind = ImageDataKind.Url, Value = GetScaledPath(uri.OriginalString, assets) };
+					}
+				}
+				catch (Exception e)
+				{
+					return new ImageData() { Kind = ImageDataKind.Error, Error = e };
+				}
+			}
+
+			private static string GetScaledPath(string path, HashSet<string> assets)
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs:86]: this logic already exists for other platforms no? I'm surprised to see a copy of it here.

File: src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+					var resolutionScale = (int)DisplayInformation.GetForCurrentView().ResolutionScale;
+
+					for (var i = KnownScales.Length - 1; i >= 0; i--)
+					{
+						var probeScale = KnownScales[i];
+					
+						if (resolutionScale >= probeScale)
+						{
+							var filePath = Path.Combine(directory, $"{filename}.scale-{probeScale}{extension}");
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs:102]: the image can set its scale like myimage.scale-100.png but also via the directory i.e. scale-100/myimage.png

Comment (MatFillion) [reply src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs:102]: https://docs.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast#reference-a-string-or-image-resource-by-name

Comment (ebariche) [reply src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs:102]: You are a 100% correct, this is the bare minimum we need to support for the gallery and will need to be revisited. This feature is significantly more complex than this : https://docs.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast

This was agreed upon between Jérôme and François.

File: src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs (hunk 1 above)
Comment (ebariche) [reply src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs:86]: See below.

**Pull request merged**